app/Providers/Common/ClientProvider.php

```diff
@@ -104,5 +108,21 @@ function (Application $app) {
                 return new GuzzleDemandClient($app->make(SignatureVerifier::class));
             }
         );
+
+        $this->app->bind(
+            Ads::class,
+            function () {
+                $drv = new CliDriver(
+                    (string)config('app.adshares_address'),
+                    (string)config('app.adshares_secret'),
+                    (string)config('app.adshares_node_host'),
+                    (int)config('app.adshares_node_port')
+                );
+                $drv->setCommand((string)config('app.adshares_command'));
+                $drv->setWorkingDir((string)config('app.adshares_workingdir'));
+
+                return new PhpAdsClient(new AdsClient($drv));
```

Comment:
AdsClient is already binded in AppServiceProvider. You can use 

/** @var Application $app */
$app->make(AdsClient::class)

Reply:
:+1: